
---
bool valid
float64 x
float64 y
